### PR TITLE
[FIX] hr_fleet: sync partner with the employee

### PR DESCRIPTION
**Pull request by yava-odoo**
In this Pr, the driver or the future driver of the car is in sync with the partner of the employee

Task-3693838

### CODE REVIEW

File: addons/hr_fleet/models/employee.py [hunk 1]
```diff
@@ -75,6 +75,15 @@ def write(self, vals):
...
         return res
 
+    def _sync_user(self, user, employee_has_image=False):
+        cars = self.env['fleet.vehicle'].search(['|', ('future_driver_id', '=', self.work_contact_id.id), ('driver_id', '=', self.work_contact_id.id)])
+        for car in cars:
+            if car.future_driver_id == self.work_contact_id:
+                car.future_driver_id = user.partner_id
+            if car.driver_id == self.work_contact_id:
+                car.driver_id = user.partner_id
+        return super()._sync_user(user, employee_has_image=False)
```
Comment (sofiagvaladze) [inline addons/hr_fleet/models/employee.py:85]: Let's not override employee_has_image argument
```suggestion
        return super()._sync_user(user, employee_has_image)
```

**Review by sofiagvaladze — CHANGES_REQUESTED** [review]
@yava-odoo the code looks good to me.. I only have little comment, also enterprise branch can be deleted. I explain there why

Can you add test for this flow? :pray:

File: addons/hr_fleet/models/employee.py (hunk 1 above)
Comment (yava-odoo) [reply addons/hr_fleet/models/employee.py]: Hello @sofiagvaladze 
I've added the test case :innocent: 
Thank you

**Comment by sofiagvaladze** [comment]
robodoo r+

**Pull request closed without merging**